### ISSUE TITLE
Moderator leaving a room with lobby enabled is permanently locked out as long as guests (and only guests) still remain in that room

### ISSUE THREAD

**Issue by drpetersen** [jitsi/jitsi-meet#7561]
### Description:

In a [secure domain setup](https://jitsi.github.io/handbook/docs/devops-guide/secure-domain) (with [LDAP authentication](https://github.com/jitsi/jitsi-meet/wiki/LDAP-Authentication#ldap-authentication-for-jitsi-meet-via-ldap2) and anonymous guest access), when the only moderator of a room enables the Lobby, but does not set a password, and subsequently leaves that room, there is no way for him/her to re-enter the room as long as there is still at least one non-moderator (guest) user in the room.

### Steps to reproduce:

1. Create a room
2. Authenticate as moderator (User A)
3. Have another non-moderator user (User B) join the room
4. Let User A enable the lobby
5. As User A drop out of the room, leaving guest User B alone in the room
6. As User A, try to enter the room

### Expected behavior:

The still (as per the browser cache) authenticated User A should be able to enter the room, in one of the following three ways (in descending order of my subjective preference):
1. When the last moderator leaves a room, the lobby is automatically disabled for that room.
2. The authentication should bypass the lobby (this might not be desirable for privacy reasons, as an authenticated user might then be able to intrude into another moderator's room)
3. When the last moderator leaves a room, one of the guest users is promoted to moderator for that room (which might be undesirable as well, e.g. because the guest user might then be granted the right to dial out with jigasi)

### Actual behavior:

User A is presented with the option to wait until someone in the room lets them in (which obviously cannot happen) or to enter the room password (which, by assumption, has not been set).

### Server information:

- Jitsi Meet version: 2.0.4950-1 (Debian package)
- Operating System: Debian 10 (buster)

### Client information:

- Browser / app version: Chromium 83.0.4103.116, Firefox 79.0 (but probably any other browser as well)
- Operating System: Debian sid/bullseye

### Additional information:

The issue has been discussed and confirmed [here](https://community.jitsi.org/t/lobby-waiting-room/27752/149?u=lpetersen).

**Comment by heikar** [comment]
I thinks it is also interesting for the "normal" authentication with username and password. If the moderator leaves the conference, no other paticipant should be automatically the "new" moderator.

**Comment by AliKarpuzoglu** [comment]
@damencho am I understanding correctly that the JID does actually change?  So the whitelisting part is not as straightforward?

**Comment by damencho** [comment]
> @damencho am I understanding correctly that the JID does actually change? So the whitelisting part is not as straightforward?

When using jwt or anonymous authentication, the jid is different on every connect.

**Comment by KIRA009** [comment]
Has their been any advancement regarding the issue?

**Comment by rfdparker** [comment]
Also setting a password when enabling the lobby largely seems to avoid this issue. Assuming the password is remembered by at least the moderator who enabled the lobby, they can enter the password to get back in while normal users aren't required to enter the password and can instead continue to wait to be 'let in' as normal.

For secure domains setups, would it make sense for the UI to require a password be set before the lobby can be enabled?

**Comment by drpetersen** [comment]
You are right, @rfdparker , having a password does help, and I am using the persistent password workaround [discussed here](https://community.jitsi.org/t/persistent-passwords-on-self-hosted-rooms/20420) for that.

The problem with setting the password _before_ enabling the lobby is that any guests arriving when the password has already been set but the lobby has not yet been enabled see the password dialog; when the lobby is enabled, the password dialog does not go away, so they do not learn that instead of the password (which they likely are not supposed to know) they could ask the moderator to admit them.

My use case is a permanent, personal meeting room (like you have in … ahem … the other videoconferencing software) which I link to in my email signature and which is protected by a permanent password (ideally never shared with anyone). When I enter that room (with my password), I activate the lobby so as to be able to admit clients (students, in my case) for consultation one by one (or group by group). If however the first client has already been waiting before I enter and enable the lobby, they see that password dialog and most probably click Cancel and are gone (and come back asking why I hadn't given them the password). Only if they dare to hit Ok with an empty (or just any) password, they get a chance to see the option to ask a moderator.

It would help a lot if instead of the modal dialog, guests would see the option to ask a moderator as soon as the lobby is created. For my setup with the permanent password, it would also help if a lobby was created automatically whenever a permanent password is in place. Although it was a pleasure to learn my first steps around Lua, I could not yet progress to the point where I could implement this myself …

> For secure domains setups, would it make sense for the UI to require a password be set before the lobby can be enabled?

Since the lobby should ideally be enabled _before_ the password, I think it might be an even better solution if, upon enabling the lobby, the password entry field opens with an auto-generated (reasonably long, like six digits) password preset, which the user can simply accept or overwrite (or even delete, if they absolutely want to).

Sorry, I am aware that this discussion might better fit in the forum, but still, I hope it is helpful to discuss some options for resolving (or at least working around) this issue.

**Comment by Keisial** [comment]
Rather than any of the three options mentioned in the initial description, I would recommend simply that when an authenticated user joins to a lobby-enabled room and there is no-one able to approve, the lobby is bypassed.
This is slightly different than (1) since the lobby itself would still be enabled. And applying to guests as well. And once the moderator rejoined, it would work normally again.

**Comment by pratik9722** [comment]
@damencho Does [it](https://github.com/jitsi/jitsi-meet/commit/b559cb8ec6dbc62d24fa816d6ec424ca6b345eeb) solve this [issue](https://github.com/jitsi/jitsi-meet/issues/7561) ?

**Comment by nicosomb** [comment]
Hello there. 
We have the same issue than @drpetersen. 

We use jitsi-token-moderation-plugin (https://github.com/nvonahsen/jitsi-token-moderation-plugin) to create a jwt token. 

If the moderator leaves (accidentally or not) the room, he/she can't come back in the room. 

We have the 2.0.5765 release (with the commit mentionned by @pratik9722), and there is still the issue.

**Comment by damencho** [comment]
> @damencho Does [it](https://github.com/jitsi/jitsi-meet/commit/b559cb8ec6dbc62d24fa816d6ec424ca6b345eeb) solve this [issue](https://github.com/jitsi/jitsi-meet/issues/7561) ?

Nope. You need some custom module that will be taking care of that. If someone is going to be promoted to be moderator add its jid to the members jids in the room by setting it to be a member like this: https://github.com/jitsi/jitsi-meet/blob/428c3cef38459e1e7ec1171f2460052d9d1d7577/resources/prosody-plugins/mod_muc_lobby_rooms.lua#L339

**Comment by veratiz** [comment]
We use jwt token authorization and this is solution for joining moderator without invite to lobby room:

```
        local whitelistJoin = false;
        local isModerator = false;

        if session.jitsi_meet_room ~= nil then
            module:log("info", "Joining moderator without invite");
            whitelistJoin = true;
            isModerator = true;
        end

        -- whitelist participants
        if whitelist:contains(invitee_domain) or whitelist:contains(invitee_bare_jid) then
            whitelistJoin = true;
        end

        local password = join:get_child_text('password', MUC_NS);
        if password and room:get_password() and password == room:get_password() then
            whitelistJoin = true;
        end

        if whitelistJoin then
            local affiliation = room:get_affiliation(invitee);
            if not affiliation or affiliation == 0 then
                if isModerator then
                    event.occupant.role = 'moderator';
                    room:set_affiliation(true, invitee_bare_jid, 'owner');
                else
                    event.occupant.role = 'participant';
                    room:set_affiliation(true, invitee_bare_jid, 'member');
                end

                room:save();

                return;
            end
        end
```

**Comment by scheibling** [comment]
In case anyone is still struggling with this, I've adapted the solution @veratiz provided above for use with mod_token_moderation or mod_token_affiliation:

https://gist.github.com/scheibling/4b2d418bb447f8953249aa3e5522c0fb

**Comment by umk0m1qk** [comment]
Three years later, this still appears to be an issue. Happened to me yesterday on an important call. I was the moderator with lobby enabled (no password). Briefly lost connectivity. Couldn't re-enter the meeting as it prompted for a password I apparently never had. Let's just say the participants were not impressed and gently chided me for not using Zoom. Any timeline to merge a permanent fix?
